nodejs/week3/app.js

```diff
@@ -0,0 +1,51 @@
+const knex = require('knex')({
+  client: 'mysql2',
+  connection: {
+    host: process.env.DB_HOST || '127.0.0.1',
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || 'root',
+    password: process.env.DB_PASSWORD || 'pass',
+    database: process.env.DB_NAME || 'hyf_node_week3_warmup',
+    multipleStatements: true
+  }
+});
+
+const express = require('express');
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use('/api', apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use('/contacts', contactsAPIRouter);
+
+contactsAPIRouter.get('/', async (req, res) => {
```

Comment:
FYI Routers are generally used to split routes into seperate files. In your case you could just create the route on the app, which is shorter:

```javascript
import express from 'express';

const app = express();
const port = process.env.PORT || 3000;

app.get('/api/contacts', (req, res) => {
  // ...
});

app.listen(port, () => {
  console.log('API server listening on port 3000');
});
```